### ISSUE TITLE
[docs] Android File System Access

### ISSUE THREAD

**Issue by Iaphetes** [tauri-apps/tauri#11733]
Hi,
currently I'm trying to access the file system in an Android 14 context.
I've already done all, that needs to be done for the Linux side, and can now access a file from within a js/html context.
However the same success eludes me in the Android 14 domain.

I've followed the instructions 
[https://v2.tauri.app/plugin/file-system/#configuration](https://v2.tauri.app/plugin/file-system/#configuration)

I also added `fs:allow-home-read-recursive` in `src-tauri/capabilities/default.json`

The path resulting from convertFileSrc seems legit.
I think the only thing missing is, that the app actually requests the permissions.

If this is a _me_ error, I would like to see a complete example showcasing android filesystem access, which I can follow start to finish to get e.g. an image from the file system on screen.

Thanks in advance.

**Comment by skorphil** [comment]
> Hi, currently I'm trying to access the file system in an Android 14 context. I've already done all, that needs to be done for the Linux side, and can now access a file from within a js/html context. However the same success eludes me in the Android 14 domain.
> 
> I've followed the instructions https://v2.tauri.app/plugin/file-system/#configuration
> 
> I also added `fs:allow-home-read-recursive` in `src-tauri/capabilities/default.json`
> 
> The path resulting from convertFileSrc seems legit. I think the only thing missing is, that the app actually requests the permissions.
> 
> If this is a _me_ error, I would like to see a complete example showcasing android filesystem access, which I can follow start to finish to get e.g. an image from the file system on screen.
> 
> Thanks in advance.

So hard to see that still nothing was changed :(

I made playground to explore plugin and wrote article: https://github.com/skorphil/tauri-fs-android-starter

TLDR: Tauri can access file only in limited number of directories without asking permissions.
There is still no plugin to request storage access.

**Comment by xcenweb** [comment]
我发现了一个全新的插件，应该能解决你的问题：[link](https://github.com/aiueo13/tauri-plugin-android-fs)

**Comment by xcenweb** [comment]
> Hi,  你好， currently I'm trying to access the file system in an Android 14 context. 目前，我正在尝试在 Android 14 上下文中访问文件系统。 I've already done all, that needs to be done for the Linux side, and can now access a file from within a js/html context. 我已经完成了所有需要在 Linux 端完成的工作，现在可以从 js/html 上下文中访问文件。 However the same success eludes me in the Android 14 domain. 然而，在 Android 14 领域，同样的成功让我望而却步。
> 
> I've followed the instructions我已经按照指示 https://v2.tauri.app/plugin/file-system/#configuration
> 
> I also added `fs:allow-home-read-recursive` in `src-tauri/capabilities/default.json`我还在 `src-tauri/capabilities/default.json` 中添加了 `fs：allow-home-read-recursive`
> 
> The path resulting from convertFileSrc seems legit.从 convertFileSrc 产生的路径似乎是合法的。 I think the only thing missing is, that the app actually requests the permissions. 我认为唯一缺少的是，应用程序实际上请求权限。
> 
> If this is a _me_ error, I would like to see a complete example showcasing android filesystem access, which I can follow start to finish to get e.g. an image from the file system on screen.如果这是_我的_错误，我希望看到一个完整的例子展示 android 文件系统访问，我可以按照开始到结束，例如从屏幕上的文件系统中获取图像。
> 
> Thanks in advance.  先谢了。

I have discovered a brand new plugin that should be able to solve your problem: [link](https://github.com/aiueo13/tauri-plugin-android-fs)
